sunpy/net/dataretriever/client.py

```diff
@@ -332,10 +341,23 @@ def fetch(self, qres, path=None, error_callback=None, **kwargs):
         qres : `~sunpy.net.dataretriever.QueryResponse`
             Results to download.
 
+        path : string or pathlib.Path
+            Path to the download directory
+
+        error_callback : Function
+            Callback function for error during downloads
+
         Returns
         -------
         Results Object
         """
+        # Check for type of path
+        if path is not None and HAS_PATHLIB and isinstance(path, pathlib.Path):
```

Comment:
You  want `if path is not None` to encompass the other if/elif

Reply:
Oops! Sorry!